app/kumactl/cmd/delete/delete_meshes.go

```diff
@@ -0,0 +1,41 @@
+package delete
+
+import (
+	"context"
+	"github.com/Kong/kuma/pkg/core/resources/apis/mesh"
+	"github.com/Kong/kuma/pkg/core/resources/model"
+	"github.com/Kong/kuma/pkg/core/resources/store"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+func newDeleteMeshCmd(pctx *deleteContext) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "mesh",
+		Short: "Delete Mesh",
+		Long:  `Delete Mesh.`,
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			rs, err := pctx.CurrentResourceStore()
+			if err != nil {
+				return err
+			}
+			meshName := args[0]
+
+			getOptions := store.GetByKey(model.DefaultNamespace, meshName, meshName)
+			meshDetails := mesh.MeshResource{}
+			if err := rs.Get(context.Background(), &meshDetails, getOptions); err != nil {
+				return errors.Wrapf(err, "failed to get Mesh with the name %s", meshName)
+			}
+
+			deleteOptions := store.DeleteByKey(model.DefaultNamespace, meshName, meshName)
+			if err := rs.Delete(context.Background(), &meshDetails, deleteOptions); err != nil {
```

Comment:
`&meshDetails` can be replaced with `&MeshResource{}` and `Get(..)` is no longer necessary